v3.7.1 and master failing to compile with spectral_double variants

I followed the compilation guide here: https://mitsuba.readthedocs.io/en/stable/src/developer_guide/compiling.html

Compilation with the default mistuba.conf file succeeds without issue. This includes variants: "scalar_rgb", "scalar_spectral", "cuda_ad_rgb", "llvm_ad_rgb", "llvm_ad_spectral"

I also was able to compile fine with "cuda_spectral" and "scalar_rgb_double" without issue.

However, when attempting to build with "cuda_spectral_double" or "scalar_spectral_double", the build fails with the following error message (same if I use master or checkout the v3.7.1 tag). This issue was not present in 3.6.4. I believe, based on the error messages, it's due to a relatively minor mistake in the SunSkyEmitter.

```
FAILED: src/emitters/CMakeFiles/timed_sunsky.dir/timed_sunsky.cpp.o 
/usr/bin/clang++-17 -DLITTLE_ENDIAN -DMI_ENABLE_AUTODIFF=1 -DMI_ENABLE_CUDA=1 -DMI_ENABLE_EMBREE=1 -DMI_ENABLE_LLVM=1 -Dtimed_sunsky_EXPORTS -I/home/dev/mitsuba-v3-7-1/include -I/home/dev/mitsuba-v3-7-1/ext/tinyformat -I/home/dev/mitsuba-v3-7-1/ext/nanobind/include -I/home/dev/mitsuba-v3-7-1/build/include -I/home/dev/mitsuba-v3-7-1/ext/embree/include -I/home/dev/mitsuba-v3-7-1/src/../ext/nanobind/ext/robin_map/include -I/home/dev/mitsuba-v3-7-1/ext/drjit/include -I/home/dev/mitsuba-v3-7-1/ext/drjit/ext/drjit-core/include -I/home/dev/mitsuba-v3-7-1/ext/drjit/ext/drjit-core/ext/lz4 -I/home/dev/mitsuba-v3-7-1/ext/drjit/ext/drjit-core/ext/nanothread/include -fcolor-diagnostics -O3 -DNDEBUG -std=gnu++17 -fPIC -fvisibility=hidden -march=native -Wall -Wextra -Wno-unused-local-typedefs -fno-math-errno -ffp-contract=fast -fno-trapping-math -MD -MT src/emitters/CMakeFiles/timed_sunsky.dir/timed_sunsky.cpp.o -MF src/emitters/CMakeFiles/timed_sunsky.dir/timed_sunsky.cpp.o.d -o src/emitters/CMakeFiles/timed_sunsky.dir/timed_sunsky.cpp.o -c /home/dev/mitsuba-v3-7-1/src/emitters/timed_sunsky.cpp
/home/dev/mitsuba-v3-7-1/src/emitters/timed_sunsky.cpp:388:29: error: no matching function for call to 'max'
  388 |                     min_w = std::max(MI_CIE_MIN, WAVELENGTHS<ScalarFloat>[0]),
      |                             ^~~~~~~~
/home/dev/mitsuba-v3-7-1/src/emitters/timed_sunsky.cpp:200:5: note: in instantiation of member function 'mitsuba::TimedSunskyEmitter<drjit::JitArray<JitBackend::CUDA, double>, mitsuba::Spectrum<drjit::JitArray<JitBackend::CUDA, double>, 4>>::sample_wlgth' requested here
  200 |     TimedSunskyEmitter(const Properties &props) : Base(props) {
      |     ^
/home/dev/mitsuba-v3-7-1/src/emitters/timed_sunsky.cpp:518:18: note: in instantiation of member function 'mitsuba::TimedSunskyEmitter<drjit::JitArray<JitBackend::CUDA, double>, mitsuba::Spectrum<drjit::JitArray<JitBackend::CUDA, double>, 4>>::TimedSunskyEmitter' requested here
  518 | MI_EXPORT_PLUGIN(TimedSunskyEmitter)
      |                  ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/bits/algorithmfwd.h:403:5: note: candidate template ignored: deduced conflicting types for parameter '_Tp' ('float' vs. 'double')
  403 |     max(const _Tp&, const _Tp&);
      |     ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/bits/stl_algo.h:5715:5: note: candidate template ignored: could not match 'initializer_list<_Tp>' against 'float'
 5715 |     max(initializer_list<_Tp> __l, _Compare __comp)
      |     ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/bits/stl_algo.h:5705:5: note: candidate function template not viable: requires single argument '__l', but 2 arguments were provided
 5705 |     max(initializer_list<_Tp> __l)
      |     ^   ~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/bits/algorithmfwd.h:408:5: note: candidate function template not viable: requires 3 arguments, but 2 were provided
  408 |     max(const _Tp&, const _Tp&, _Compare);
      |     ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/dev/mitsuba-v3-7-1/src/emitters/timed_sunsky.cpp:389:29: error: no matching function for call to 'min'
  389 |                     max_w = std::min(MI_CIE_MAX, WAVELENGTHS<ScalarFloat>[WAVELENGTH_COUNT - 1]);
      |                             ^~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/bits/algorithmfwd.h:416:5: note: candidate template ignored: deduced conflicting types for parameter '_Tp' ('float' vs. 'double')
  416 |     min(const _Tp&, const _Tp&);
      |     ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/bits/stl_algo.h:5695:5: note: candidate template ignored: could not match 'initializer_list<_Tp>' against 'float'
 5695 |     min(initializer_list<_Tp> __l, _Compare __comp)
      |     ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/bits/stl_algo.h:5685:5: note: candidate function template not viable: requires single argument '__l', but 2 arguments were provided
 5685 |     min(initializer_list<_Tp> __l)
      |     ^   ~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/bits/algorithmfwd.h:421:5: note: candidate function template not viable: requires 3 arguments, but 2 were provided
  421 |     min(const _Tp&, const _Tp&, _Compare);
      |     ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
```

--- a/src/emitters/timed_sunsky.cpp
+++ b/src/emitters/timed_sunsky.cpp
@@ -385,8 +385,8 @@ private:
     std::pair<Wavelength, Spectrum> sample_wlgth(const Float& sample, Mask active) const override {
         if constexpr (is_spectral_v<Spectrum>) {
             static constexpr ScalarFloat
-                    min_w = std::max(MI_CIE_MIN, WAVELENGTHS<ScalarFloat>[0]),
-                    max_w = std::min(MI_CIE_MAX, WAVELENGTHS<ScalarFloat>[WAVELENGTH_COUNT - 1]);
+                    min_w = std::max(ScalarFloat(MI_CIE_MIN), WAVELENGTHS<ScalarFloat>[0]),
+                    max_w = std::min(ScalarFloat(MI_CIE_MAX), WAVELENGTHS<ScalarFloat>[WAVELENGTH_COUNT - 1]);
 
             Wavelength wavelengths = math::sample_shifted<Wavelength>(sample);
             wavelengths = min_w + (max_w - min_w) * wavelengths;

